sys/net/gnrc/network_layer/ipv6/nib/_nib-6lr.c

```diff
@@ -84,8 +84,16 @@ uint8_t _reg_addr_upstream(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
 #endif  /* CONFIG_GNRC_IPV6_NIB_MULTIHOP_DAD */
             if (aro->ltime.u16 != 0) {
                 _handle_sl2ao(netif, ipv6, icmpv6, sl2ao);
+
                 /* re-get NCE in case it was updated */
                 nce = _nib_onl_get(&ipv6->src, netif->pid);
+
+                /* NIB is full */
+                /* TODO: evict oldest entry */
```

Comment:
`_nib_onl_get()` should already do that if the cache entries are garbage collectible (which they most likely are not on a 6LR). See also https://tools.ietf.org/html/rfc6775#section-3.5.

Reply:
Ping? This misleading todo comment should be removed.

Reply:
Wouldn't it still be preferable to drop an old entry that *might* already be obsolete compared to dropping a new entry where we *know* that it is fresh?

Reply:
If the entry is obsolete its registration lifetime would run out, making the entry garbage collectible again. We should not break 6Lo duplicate address detection over this, when there are viable ways out of it. Either increase the neighbor cache on your routers to accommodate for your network or let the registering nodes register with a different router, balancing the network more. The alternative to not registering would be to not use 6Lo-ND, but that activates more multicast within the network.

Reply:
Side-note: technically, your sleepy node behaves faulty, since it changes its EUI-64. This way ND does not have any way to re-identify the node and thus recognizes it as a new node.

Reply:
I know, and there is already a PR to fix it #15562 :wink: 
But there might as well be many sleepy nodes - but you are right that a random new node appearing should not break connectivity for existing nodes.
I thought if an existing node is dropped, it would simply re-negotiate a prefix the next time it wants to communicate with the border router, but I suppose it's not that simple.

Reply:
> I thought if an existing node is dropped, it would simply re-negotiate a prefix the next time it wants to communicate with the border router, but I suppose it's not that simple.

Usually it does, but only if the address does not change and if the node makes sure too keep their dead-lines (address registration mostly).

Reply:
> Usually it does, but only if the address does not change

eh, don't get too hung up about this - just imagine the sleepy node would not restart, but go to sleep for a day and another sleepy node appears instead - that's much closer to reality anyway

Reply:
The registration time is provided in minutes and has 16-bit width. This gives plenty of space (~45 days) to configure the node to tell the border router that you won't be available for a day or more ;-). Just configure [the registration time macro](https://doc.riot-os.org/group__net__gnrc__sixlowpan__config.html#ga15dd2e19541dd74baf294e9c22f47c7e) to the desired value :-)

Reply:
That sounds convenient if I want to reduce start-up latency for long sleeping nodes.
But now I wonder - shouldn't I rather set this to the expected wake time of a node if there are more nodes than fit the neighbor cache of the border router, given that they won't all be awake at the same time?

Reply:
First, just to be clear: That registration time is set by the node (who knows best when they will wake up), not the BR.

> But now I wonder - shouldn't I rather set this to the expected wake time of a node if there are more nodes than fit the neighbor cache of the border router, given that they won't all be awake at the same time?

You should it set slightly longer of course, otherwise you might miss your deadline ;-). The Neighbor Cache is "misused" in 6Lo-ND as a address registry. So if you can _absolutely_ guarantee, that they have different addresses or that only nodes that are awake at different times have the same address (which technically you only really can via the address registry, unless you hand-configure every nodes address), you can do such a thing. Otherwise, you run into the risk of having addresses with the same address in your network, which can lead to a whole different set of confusion ;-).